File: worker/src/main/java/ru/variantum/worker/repository/AnalysisEventRepository.java
package ru.variantum.worker.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.variantum.worker.domain.AnalysisEventRecord;

import java.util.List;
import java.util.UUID;

public interface AnalysisEventRepository extends JpaRepository<AnalysisEventRecord, UUID> {

    List<AnalysisEventRecord> findTop20ByOrderByReceivedAtDesc();
}

File: worker/src/main/kotlin/ru/variantum/worker/repository/AnalysisEventRepository.kt
package ru.variantum.worker.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.variantum.worker.domain.AnalysisEventRecord
import java.util.UUID

interface AnalysisEventRepository : JpaRepository<AnalysisEventRecord, UUID> {

    fun findTop20ByOrderByReceivedAtDesc(): List<AnalysisEventRecord>
}
